frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from "react";
+import type { Dispatch, SetStateAction } from "react";
+import { Icon } from "metabase/core/components/Icon";
+import { Group, Menu, Stack, Text } from "metabase/ui";
+import { isEmpty } from "metabase/lib/validate";
+import type { SelectedComparison } from "metabase-types/api";
+import { COMPARISON_TYPES } from "../constants";
+import type { ComparisonMenuOption } from "../types";
+import { PeriodsAgoOptionComponent } from "./PeriodsAgoOptionComponent";
+import { ButtonStyled } from "./SmartScalarSettingsWidgets.styled";
+import { MenuItemStyled } from "./MenuItem.styled";
+
+type SmartScalarComparisonWidgetProps = {
+  onChange: (setting: { type: string; value?: number }) => void;
+  options: ComparisonMenuOption[];
+  value: SelectedComparison;
+};
+
+export function SmartScalarComparisonWidget({
+  onChange,
+  options,
+  value: selectedValue,
+}: SmartScalarComparisonWidgetProps) {
+  const [open, setOpen] = useState(false);
+
+  const selectedOption = options.find(
+    ({ type }) => type === selectedValue.type,
+  );
+
+  const selectedDisplayName =
+    selectedValue.type === COMPARISON_TYPES.PERIODS_AGO
+      ? `${selectedValue.value ?? ""} ${selectedOption?.name}`
+      : selectedOption?.name;
+
+  const isDisabled = options.length === 1 && !isEmpty(selectedOption);
+
+  return (
+    <Menu opened={open} onChange={setOpen} position="bottom-start" shadow="sm">
+      <Menu.Target>
+        <ButtonStyled
+          data-testid={"comparisons-widget-button"}
+          disabled={isDisabled}
+        >
+          <Group spacing="sm">
+            {selectedDisplayName}
+            <Icon name="chevrondown" size="14" />
+          </Group>
+        </ButtonStyled>
```

Comment:
Is there a reason we're not using `Button's` `rightIcon` prop?

Reply:
Because how the label and `rightIcon` get separated, I do not know how to get the styles right using the `rightIcon` attribute

![image](https://github.com/metabase/metabase/assets/22608765/4248370f-c766-4680-8864-f511de7fdc2b)

Reply:
For now, I switched to using `rightIcon` and setting `justify-content: center` with inline styles, because that is the only pattern that has support across Mantine v6 -> v7. Alternatively we can target the child div for now and make that have `justify-content: center`. 

We can discuss this in a follow up PR.